### PR TITLE
[USER] [PAGE] Updatehomereport

### PR DESCRIPTION
**Pull request by antoniusfilian**
update home report with data

image

### CODE REVIEW

File: src/app/app.component.html [hunk 1]
```diff
@@ -1 +1,9 @@
+<!-- <page-admin-report-detail [dataDummy]="dummyReportDetail"></page-admin-report-detail> -->
+
+<!-- <add-report [dataProjectName] = "dummyProjectName" [dataRoles] = "dummyRoles"></add-report> -->
+<!-- <app-login-scene></app-login-scene> -->
```
Comment (danielindracahyadi) [inline src/app/app.component.html]: please remove all of the comments

File: src/app/app.component.ts [hunk 2]
```diff
@@ -581,6 +581,238 @@ export class AppComponent {
       dataDropdown : 'User',
     }
   ];
+  @Output() homeData =
+  [
+     {
+        success: true,
+        data:
```
Comment (danielindracahyadi) [inline src/app/app.component.ts]: Please move all of the homeData to homeService, for all of the state stored inside of it. Doing so will ease you in the future.

File: src/app/core-ui/custom-text-view/custom-text-view.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (danielindracahyadi) [inline src/app/core-ui/custom-text-view/custom-text-view.component.spec.ts]: remove this

File: src/app/core-ui/dropdown/dropdown.component.html [hunk 4]
```diff
@@ -1,6 +1,16 @@
...
     <option [ngValue]="undefined" hidden>{{placeholder}}</option>
     <option *ngFor="let data of result">{{data.dataDropdown}} </option>
   </select>
 </div>
+  
+<!-- 
+<mat-form-field class={{classtouse}}>
+  <mat-label>{{placeholder}}</mat-label>
+  <mat-select (selectionChange)="selected()" [(ngModel)]="selectedData">
+    <mat-option *ngFor="let data of result">
```
Comment (danielindracahyadi) [inline src/app/core-ui/dropdown/dropdown.component.html]: please remove the comment

File: src/app/pages/home-report/home-report.component.ts [hunk 5]
```diff
@@ -9,7 +9,10 @@ export class HomeReportComponent implements OnInit {
 
   constructor() { }
 
+  @Input() dataHome = '';
+
   ngOnInit() {
+    // console.log(this.dataHome);
```
Comment (danielindracahyadi) [inline src/app/pages/home-report/home-report.component.ts]: remove this

File: src/app/app.component.ts (hunk 2 above)
Comment (antoniusfilian) [reply src/app/app.component.ts]: it means that homeData cant be accessed from app.component.html @danielindracahyadi

Comment (danielindracahyadi) [reply src/app/app.component.ts]: you need services for communicating between components

File: src/app/app.service.ts [hunk 6]
```diff
@@ -60,6 +60,243 @@ export class AppService {
     selected: ''
   };
 
+  private homeData =
+  [
```
Comment (danielindracahyadi) [inline src/app/app.service.ts]: no need to be array, make it object, response only returns one object

Comment (antoniusfilian) [reply src/app/app.service.ts]: okay daniel, thank you @danielindracahyadi

**Pull request merged**